### DISCUSSION TITLE
Document the key used to sign the phar

### DISCUSSION THREAD

**Discussion by lucc** [composer-unused/composer-unused#601] [Ideas]
It seems that a new key is used to sign the phar.  There is even a ci job to check it but I could not find it documented in the readme or the changelog.

The old key was F4D32E2C9343B2AE, the new key seems to be 3026B3ADAD51C8E74122DA1FB1B5F6D21AB070E1.

I would like the key to be documented in some official place so that I can tell phive which key to accept.  At the moment I just have to tell it to accept whatever key is coming along and have no (more or less Independent) validation that it is the right key.

**Comment by icanhazstring** [comment]
The old key was old and lost on my old Mac.
So I had to create a new one, which I uploaded to normal public key ring servers containing the same mails as before.

But what do you have in mind like documenting it. Because I haven't seen something like that around. Maybe you got some examples I can take here?

**Comment by lucc** [comment]
You could just add a sentence to the readme.  I reported the same problem at phpmd, my argument is [here](https://github.com/phpmd/phpmd/issues/1030#issuecomment-1742685581).

They resolved it here: https://github.com/phpmd/phpmd/pull/1041

**Reply by icanhazstring** [reply]
Thanks for the reference. Will add something along the line. Thank you for raising this 👍

**Comment by lucc** [comment]
Hello @icanhazstring did you ever come around to document the key? It seems it changed again.  I am again seeing failing CI jobs on my end and I can not find any reference in this repository which the correct key should be.

**Reply by icanhazstring** [reply]
The PHAR built can be ignored in forks. The key can't be directly used as it is password protected.

**Reply by lucc** [reply]
Sorry I do not understand that response. I am not clonig your repo. I try to download `composer-unused` with phive into my own ci pipeline  (for some totally unrelated php project).

I am not trying to use your key (that should be impossible anyways because it would effectifly leak your key).  I am trying to verify the signature made with you key but I do not know if the signature was made with the correct key or if some man in the middle change the phar and made a new signature with their key.

**Reply by icanhazstring** [reply]
Ah sorry. Misunderstanding from my side.
I had to renew my key. But I already uploaded my key to multiple key servers.

I have to recheck why phive is having problems here.

**Reply by icanhazstring** [reply]
But I just saw, that phive supports certain keys. I can give you the key again and document it. Yes.

**Reply by lucc** [reply]
Phive is not having any problems. I am having a problem but it is not technical but a problem about documentation.

You are using a key but knowbody told me that it is your key.  I would like to ask you to mention your key ID in the documentation of composer-unused (preferably somewhere around the installation notes).  I can obviously just tell phive to accept this key but it would be much better if the official docs of composer-unused did mention they key.  Otherwise I will have to tell phive to accept this random key that it finds when downloading composer-unused.  If some man in the middle did swap the phar and the signnature phive will report some other key that is still unknown to me but I have now idea if I am currently seeing your new key or some MID key.

Another advantage of publishing your key ID in the installation docs would be that I can write my CI script with the correct phive command and trusted key id.  Now I have to download the package once (eather by hand or, even more cumbersome in CI) just to find the key ID and the update my CI script.

**Reply by lucc** [reply]
PS: that is the same argument that I linked in my comment above, a little more than a year ago: https://github.com/composer-unused/composer-unused/discussions/601#discussioncomment-7747241

**Reply by icanhazstring** [reply]
Sorry for the whole confusion. I added the key in the installation advice.
https://github.com/composer-unused/composer-unused/blob/main/README.md#phar-php-archive-recommended

**Comment by lucc** [comment]
@icanhazstring thank you that solves it.
